<?xml version="1.0" encoding="utf-8"?>

<launch>

    <!-- # gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <!-- <arg name="world_name" value="$(find docking_gazebo)/worlds/empty.world" /> -->
        <arg name="world_name" value="$(find docking_gazebo)/worlds/cloister.world" />
        <!-- <arg name="world_name" value="$(find docking_gazebo)/worlds/shelf.world" /> -->
        <!-- <arg name="world_name" value="$(find docking_gazebo)/worlds/playpen-3.world" /> -->
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!-- # load robot description -->
    <param name="robot_description" command="$(find xacro)/xacro '$(find docking_gazebo)/urdf/big_car.xacro'" />

    <node name="spawner_model" pkg="gazebo_ros" type="spawn_model"
          args="-x 0
                -y 0
                -z 0.24
                -R 0
                -P 0
                -Y 0
                -urdf
                -model robot
                -param robot_description" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" respawn="false" output="screen" />

    <!-- # static tf -->
    <node pkg="docking_gazebo" type="br_big_car.py" name="big_car_tf_br" output="screen">
        <param name="car1_imu" value="/car1_imu" />
        <param name="car2_imu" value="/car2_imu" />
    </node>

    <!-- # merge scan, delay 2 sec -->
    <include file="$(find timed_roslaunch)/launch/timed_roslaunch.launch">
        <arg name="time" value="5" />
        <arg name="pkg" value="docking_gazebo" />
        <arg name="file" value="_merge_scan.launch" />
    </include>

</launch>
